sphinx_gallery/gen_gallery.py

```diff
@@ -687,6 +692,150 @@ def write_computation_times(gallery_conf, target_dir, costs):
             fid.write(hline)
 
 
+def write_api_entry_usage(gallery_conf, target_dir):
+    if gallery_conf['backreferences_dir'] is None:
+        return
+
+    backreferences_dir = os.path.join(gallery_conf['src_dir'],
+                                      gallery_conf['backreferences_dir'])
+    example_files = [example for example in os.listdir(backreferences_dir)
+                     if '__' not in example and
+                     ((example.endswith('.examples') and
+                       not os.path.isfile(example + '.new')) or
+                     example.endswith('.examples.new'))]
+
+    def get_entry(entry):
+        """Remove all trailing .examples and .examples.new instances."""
+        if entry.endswith('.new'):
+            entry = entry[:-4]
+        if entry.endswith('.examples'):
+            entry = entry[:-9]
+        return entry
+
+    def get_entry_type(entry):
+        """Infer type from capitalization."""
+        if any([char.isupper() for char in entry.split('.')[-1]]):
+            return 'class'
+        return 'py:obj'
+
+    # modules have classes and functions in them, so check if there exists
+    # classes or functions that have the module as a root and, if so,
+    # remove the module (it must be lower case, classes have methods
+    # that include the class name as the prefix but should be kept)
+    for example in example_files.copy():
+        if any([char.isupper() for char in example]):
+            continue  # include classes (camel case)
+        for example2 in example_files:
+            if example != example2 and \
+                    get_entry(example) in get_entry(example2):
+                example_files.remove(example)
+                break
```

Comment:
this seems brittle and maybe even wrong. `numpy.ndarray` is a class that is all lowercase, and has methods. won't this remove `ndarray`?  Is there no other way to get entry type other than inferring from capitalization?

Reply:
It does look like there is functionality to infer classes but I think it would involve fairly major refactoring of the `sphinx_gallery.backreferences.identify_names` code which is quite complex. Maybe this information could be stored in the example file though so that it doesn't have to be found twice. Either way, all you get is the string without refactoring more of the codebase and I agree that this is not great but it's part of the code I'm wary to touch because of its complexity.

Reply:
I think maybe best just to remove this... I'll go ahead and add that with a to do